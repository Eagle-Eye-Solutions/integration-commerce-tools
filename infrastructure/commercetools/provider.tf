provider "commercetools" {
  api_url       = var.CTP_API_URL
  token_url     = var.CTP_AUTH_URL
  client_id     = var.CTP_CLIENT_ID
  client_secret = var.CTP_CLIENT_SECRET
  project_key   = var.CTP_PROJECT_KEY
  scopes        = var.CTP_SCOPES
}

variable "CTP_API_URL" {
  type = string
  default = "https://api.europe-west1.gcp.commercetools.com"
  description = "commercetools API URL"
}

variable "CTP_AUTH_URL" {
  type = string
  default = "https://auth.europe-west1.gcp.commercetools.com"
  description = "commercetools Authentication API URL"
}

variable "CTP_CLIENT_ID" {
  type = string
  description = "commercetools API Client ID"
  sensitive = true
}

variable "CTP_CLIENT_SECRET" {
  type = string
  description = "commercetools API Client secret"
  sensitive = true
}

variable "CTP_PROJECT_KEY" {
  type = string
  description = "commercetools API Client ID"
}

variable "CTP_SCOPES" {
  type = string
  description = "commercetools Client scopes"
}

variable "CPT_PROJECT_NAME" {
  type = string
  description = "commercetools environment project name"
}
