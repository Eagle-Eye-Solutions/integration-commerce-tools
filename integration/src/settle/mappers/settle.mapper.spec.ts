import { Test, TestingModule } from '@nestjs/testing';
import { SettleMapper } from './settle.mapper';
import { ConfigService } from '@nestjs/config';
import { BASKET_STORE_SERVICE } from '../../common/services/basket-store/basket-store.provider';
import { BasketStoreService } from '../../common/services/basket-store/basket-store.interface';

describe('CTCartToEEBasketMapper', () => {
  let service: SettleMapper;
  let configService: ConfigService;
  let basketStoreService: jest.Mocked<BasketStoreService>;

  const shippingMethodMapMock = [{ key: 'standard-key', upc: '245879' }];

  beforeEach(async () => {
    const module: TestingModule = await Test.createTestingModule({
      providers: [
        SettleMapper,
        {
          provide: ConfigService,
          useValue: {
            get: jest.fn(),
          },
        },
        {
          provide: BASKET_STORE_SERVICE,
          useValue: {
            save: jest.fn(),
            get: jest.fn(),
            delete: jest.fn(),
            isEnabled: jest.fn(),
          },
        },
      ],
    }).compile();

    service = module.get<SettleMapper>(SettleMapper);
    configService = module.get<ConfigService>(ConfigService);
    basketStoreService = module.get(BASKET_STORE_SERVICE);
    jest.spyOn(configService, 'get').mockReturnValueOnce(shippingMethodMapMock);
    jest.resetAllMocks();
  });

  describe('mapOrderToWalletSettlePayload', () => {
    it('should return the payload for /wallet/settle', async () => {
      jest
        .spyOn(configService, 'get')
        .mockReturnValueOnce('outlet1')
        .mockReturnValueOnce('banner1');

      jest.spyOn(basketStoreService, 'get').mockResolvedValueOnce({
        enrichedBasket: {
          contents: [],
        },
      });

      const payload = await service.mapOrderToWalletSettlePayload({
        id: '123456',
        cart: {
          typeId: 'cart',
          id: '12345678',
        },
        custom: {
          fields: {},
        },
      } as any);

      expect(payload).toMatchSnapshot();
    });

    it('should include eagleeye-identityValue for /wallet/settle when available', async () => {
      jest
        .spyOn(configService, 'get')
        .mockReturnValueOnce('outlet1')
        .mockReturnValueOnce('banner1');

      jest.spyOn(basketStoreService, 'get').mockResolvedValueOnce({
        enrichedBasket: {
          contents: [],
        },
      });

      const payload = await service.mapOrderToWalletSettlePayload({
        id: '123456',
        cart: {
          typeId: 'cart',
          id: '12345678',
        },
        custom: {
          fields: {
            'eagleeye-identityValue': 'some-identity',
          },
        },
      } as any);

      expect(payload).toMatchSnapshot();
    });
  });
});
