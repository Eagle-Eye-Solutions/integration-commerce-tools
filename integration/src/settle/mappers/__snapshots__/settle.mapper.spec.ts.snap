// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CTCartToEEBasketMapper mapOrderToWalletSettlePayload should include eagleeye-identityValue for /wallet/settle when available 1`] = `
{
  "basket": {
    "contents": [],
  },
  "identity": {
    "identityValue": "some-identity",
  },
  "location": {
    "incomingIdentifier": "outlet1",
    "parentIncomingIdentifier": "banner1",
  },
  "mode": "ACTIVE",
  "reference": "12345678",
}
`;

exports[`CTCartToEEBasketMapper mapOrderToWalletSettlePayload should return the payload for /wallet/settle 1`] = `
{
  "basket": {
    "contents": [],
  },
  "location": {
    "incomingIdentifier": "outlet1",
    "parentIncomingIdentifier": "banner1",
  },
  "mode": "ACTIVE",
  "reference": "12345678",
}
`;
